v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/impl/network/handlers/packet/ActionBarEventPacketHandlers.java

```diff
@@ -17,29 +15,24 @@ public static void registerHandlers() {
         DenizenNetworkManagerImpl.registerPacketHandler(ClientboundSetActionBarTextPacket.class, ActionBarEventPacketHandlers::processActionbarPacket);
     }
 
-    public static Packet<ClientGamePacketListener> processActionbarPacket(DenizenNetworkManagerImpl networkManager, Packet<ClientGamePacketListener> packet) {
+    public static ClientboundSetActionBarTextPacket processActionbarPacket(DenizenNetworkManagerImpl networkManager, ClientboundSetActionBarTextPacket actionbarPacket) {
         if (!PlayerReceivesActionbarScriptEvent.instance.loaded) {
-            return packet;
+            return actionbarPacket;
         }
-        if (packet instanceof ClientboundSetActionBarTextPacket) {
-            ClientboundSetActionBarTextPacket actionbarPacket = (ClientboundSetActionBarTextPacket) packet;
-            PlayerReceivesActionbarScriptEvent event = PlayerReceivesActionbarScriptEvent.instance;
-            Component baseComponent = actionbarPacket.getText();
-            event.reset();
-            event.message = new ElementTag(FormattedTextHelper.stringify(Handler.componentToSpigot(baseComponent)));
-            event.rawJson = new ElementTag(Component.Serializer.toJson(baseComponent));
-            event.system = new ElementTag(false);
-            event.player = PlayerTag.mirrorBukkitPlayer(networkManager.player.getBukkitEntity());
-            event = (PlayerReceivesActionbarScriptEvent) event.triggerNow();
-            if (event.cancelled) {
-                return null;
-            }
-            if (event.modified) {
-                Component component = Handler.componentToNMS(event.altMessageDetermination);
-                ClientboundSetActionBarTextPacket newPacket = new ClientboundSetActionBarTextPacket(component);
-                return newPacket;
-            }
+        PlayerReceivesActionbarScriptEvent event = PlayerReceivesActionbarScriptEvent.instance;
```

Comment:
should probably put this before the `if`

Reply:
Done